### PR TITLE
fix(mme): Handling PDU session establishment reject Scenerios

### PR DESCRIPTION
**Pull request by ganeshg87**
test cases:
   - added UT code to validate SM Message
   - verified PDU session establishment reject scenerios using UERANSIM

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp [hunk 1]
```diff
@@ -616,4 +624,133 @@ int amf_smf_handle_ip_address_response(
...
+ **                                                                        **
+ ***************************************************************************/
+int amf_pdu_session_establishment_reject(
+    amf_ue_ngap_id_t ue_id, uint8_t session_id, uint8_t pti, uint8_t cause) {
+  nas5g_error_code_t rc  = M5G_AS_SUCCESS;
+  uint32_t bytes         = 0;
+  uint32_t len           = 0;
+  uint32_t container_len = 0;
+  bstring buffer;
+  smf_context_t* smf_ctx         = nullptr;
```
Comment (aswin-alexander) [inline lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp:648]: Where is this used?

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp [hunk 2]
```diff
@@ -616,4 +624,133 @@ int amf_smf_handle_ip_address_response(
...
+ **                session_id		: PDU Session Inputs               **
+ **		   pti              : Procedure transaction identity       **
+ **		   5GSM cause       : 5GSM cause                           **
+ **                                                                        **
+ **  Return     :  RETURNok, RETURNerror                                   **
+ **                                                                        **
+ ***************************************************************************/
+int amf_pdu_session_establishment_reject(
+    amf_ue_ngap_id_t ue_id, uint8_t session_id, uint8_t pti, uint8_t cause) {
+  nas5g_error_code_t rc  = M5G_AS_SUCCESS;
```
Comment (aswin-alexander) [inline lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp:643]: Should we not initialize to a failure and only have it go to success via amf_app_handle_nas_dl_req?

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp [hunk 3]
```diff
@@ -616,4 +624,133 @@ int amf_smf_handle_ip_address_response(
...
+  buffer = bfromcstralloc(len, "\0");
+  bytes  = nas5g_message_encode(
+      buffer->data, &msg, len, &ue_context->amf_context._security);
+  if (bytes > 0) {
+    buffer->slen = bytes;
+    amf_app_handle_nas_dl_req(ue_id, buffer, rc);
+
+  } else {
+    OAILOG_WARNING(LOG_AMF_APP, "NAS encode failed \n");
+    bdestroy_wrapper(&buffer);
```
Comment (aswin-alexander) [inline lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp:751]: Woulnt we want to clean up in the success case? after the amf_app_handle_nas_dl_req?

**Comment by ganeshg87** [comment]
@aswin-alexander, 
PR code changes in progress, not finished yet, I have jumped to another activity. I moved this PR into draft mode.

I will address your review comments.

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp (hunk 2 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp:643]: addressed review comment

File: lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp (hunk 3 above)
Comment (ganeshg87) [reply lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp:751]: addressed review comment

**Comment by ganeshg87** [comment]
#9111 PR dropped, 
new PR raised to handle above cases
https://github.com/magma/magma/pull/9813

**Pull request closed without merging**